### ISSUE TITLE
How to evaluate using the saved checkpoint?

### ISSUE THREAD

**Issue by lizekang** [studio-ousia/luke#25]
Hi, I wonder how to evaluate using the saved checkpoint after training. 
And I meet TypeError when I train Named Entity Recognition on CoNLL-2003 Dataset:
```
python -m examples.cli --model-file=luke_large_500k.tar.gz --output-dir=<OUTPUT_DIR> ner run --data-dir=<DATA_DIR> --fp16 --train-batch-size=2 --gradient-accumulation-steps=2 --learning-rate=1e-5 --num-train-epochs=5
```
The error is :
```
Eval: 100%|##########| 181/181 [04:43<00:00,  1.56s/it]
Traceback (most recent call last):
  File "/data1/zekangli/anaconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/data1/zekangli/anaconda3/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/apdcephfs/share_47076/zekangli/multilingual/luke/examples/cli.py", line 135, in <module>
    cli()
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/apdcephfs/share_47076/zekangli/multilingual/luke/examples/utils/trainer.py", line 32, in wrapper
    return func(*args, **kwargs)
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/apdcephfs/share_47076/zekangli/multilingual/luke/examples/ner/main.py", line 90, in run
    results.update({f"dev_{k}": v for k, v in evaluate(args, model, "dev", dev_output_file).items()})
  File "/apdcephfs/share_47076/zekangli/multilingual/luke/examples/ner/main.py", line 155, in evaluate
    print(seqeval.metrics.classification_report(final_labels, final_predictions, digits=4))
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/seqeval/metrics/sequence_labeling.py", line 697, in classification_report
    suffix=suffix
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/seqeval/metrics/sequence_labeling.py", line 139, in precision_recall_fscore_support
    extract_tp_actual_correct=extract_tp_actual_correct
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/seqeval/metrics/v1.py", line 122, in _precision_recall_fscore_support
    check_consistent_length(y_true, y_pred)
  File "/data1/zekangli/anaconda3/lib/python3.7/site-packages/seqeval/metrics/v1.py", line 97, in check_consistent_length
    raise TypeError('Found input variables without list of list.')
TypeError: Found input variables without list of list.
The number of labels: 51362
```

**Comment by ikuyamada** [comment]
Hi, 
The error is generated from the `seqeval` library which is used to compute the evaluation scores.
https://github.com/chakki-works/seqeval/blob/29a0e1e44f160cfa622b2ebfc59fd9aa6639f84f/seqeval/metrics/v1.py#L97

The error may be related to the input dataset. It seems that there are two formats, IOB1 and IOB2, available for the CoNLL 2003 dataset, and the code only supports IOB1 format.

**Comment by lizekang** [comment]
Thanks! After I train the model and save the checkpoint, how can I do the evaluation on the test set using the saved checkpoint?

**Comment by ikuyamada** [comment]
Hi, The evaluation on the test set should be automatically executed after the training.

**Comment by lizekang** [comment]
Hi, but if I want to evaluate other data, how can I use the saved checkpoint?

**Comment by ikuyamada** [comment]
The NER example is provided for reproducing our results on the CoNLL-2003 dataset. Please modify the code for such use case.

**Comment by ikuyamada** [comment]
I close this issue as there is no activity. Please reopen the issue if there are any further issues.

**Comment by BeyonderXX** [comment]
Oh no !

**Comment by BeyonderXX** [comment]
> The NER example is provided for reproducing our results on the CoNLL-2003 dataset. Please modify the code for such use case.

Oh no ~

**Comment by ikuyamada** [comment]
I reviewed the source code again, and noticed that there is the `--no-train` option that enables to evaluate the model performance loaded from a checkpoint file. When using this option, you need to specify `--model-file`, which is the original pretrained model file, and `--output-dir` that contains the checkpoint file.

`python -m examples.cli --model-file=luke_large_500k.tar.gz --output-dir=<OUTPUT_DIR> ner run --no-train`

**Comment by ikuyamada** [comment]
Also, as mentioned in [another issue](https://github.com/studio-ousia/luke/issues/34#issuecomment-743219480), the error mentioned in the original issue (`TypeError: Found input variables without list of list.`) can be fixed using the version 0.0.12 of seqeval library used in our original experiments.
https://github.com/studio-ousia/luke/blob/master/poetry.lock#L792
